Assets/Scripts/Backend/SpawnObject.cs

```diff
@@ -48,16 +50,22 @@ private void PlaceBuilding()
     {
         if (_building)
         {
```

Comment:
Something you can practice is to have loosely coupled if-statements. Now everything is dependent on this if-statement. You could change it to:
```
if (!building)
    return;

MyOtherCode();
```
This makes it more readable and easier to extend.